src/actions/ast.js

```diff
@@ -34,13 +34,15 @@ export function setSourceMetaData(sourceId: SourceId) {
     }
 
     const framework = await getFramework(source.id);
-    dispatch({
+    const action: Action = {
       type: "SET_SOURCE_METADATA",
       sourceId: source.id,
       sourceMetaData: {
         framework
       }
-    });
+    };
+
+    dispatch(action);
```

Comment:
we might be able to use [this](https://github.com/devtools-html/debugger.html/blob/master/src/reducers/sources.js#L45) strategy:

```js
dispatch({
  type: "SET_SOURCE_METADATA",
  sourceId: source.id,
  sourceMetaData: {
    framework
  }
}: Action);
```

Reply:
I started with that but prettier(?) wanted it to be wrapped with parenthesis and indented which seemed less readable than extracting it to variable.

```
    dispatch(
      ({
        type: "SET_SOURCE_METADATA",
        sourceId: source.id,
        sourceMetaData: {
          framework
        }
      }: Action)
    );
```

Reply:
Hmm, i think I'd opt for this... having a separate var feels awkward. Another option is exporting a dispatch function from `actions/index` that asserts the param is Action...

Reply:
:+1: I'll update those instances to the inline type